lib/ruby_lsp/listener.rb

```diff
@@ -69,14 +70,22 @@ def response
     end
 
     sig do
-      abstract.params(extension: RubyLsp::Extension).returns(T.nilable(RubyLsp::Listener[ResponseType]))
+      abstract.params(extension: RubyLsp::Extension).returns(T.nilable(RubyLsp::ExtensionListener[ResponseType]))
     end
     def initialize_external_listener(extension); end
+  end
+  private_constant(:ExtensibleListener)
 
-    # Does nothing by default. Requests that accept extensions should override this method to define how to merge
-    # responses coming from external listeners
-    sig { abstract.params(other: Listener[T.untyped]).returns(T.self_type) }
-    def merge_response!(other)
-    end
+  class ExtensionListener < Listener
+    extend T::Sig
+    extend T::Helpers
+    extend T::Generic
+
+    ResponseType = type_member
+
+    abstract!
+
+    sig { abstract.params(current_response: ResponseType).void }
+    def merge_response(current_response); end
```

Comment:
To features like `SemanticHighlighting` or `DocumentSymbol`, it will be more like updating the request with additional information instead of merging a separate response. Perhaps we should use a different verb here?

Reply:
Good point.

* `handle_response`
* `process_response`
* `decorate_response`
* `extend_response`

?

Reply:
I find merge to be intuitive, but I don't have strong opinions. Which one do people find the best?

Reply:
I like `decorate_response` and `extend_response` as they both indicate that we are augmenting the original response.